src/schema/resolvers/group.js

```diff
@@ -11,7 +12,7 @@ const resolvers = {
     group: groupController.getGroupById
   },
   Mutation: {
-    createGroup: groupController.createGroup
+    createGroup: (obj, params, context) => requireAuth(obj, params, context)(groupController.createGroup)
```

Comment:
Let's refactor this to achieve more simplicity: `requireAuth(groupController.createGroup)`

Reply:
@abelosorio  but how `requireAuth` access to `context` without passing parameters?

Reply:
When you do this:
```javascript
createGroup: requireAuth(groupController.createGroup)
```

The _output_ of `requiredAuth` will be called like any other controller ;) So the magic there, is that requiredAuth should return a function:

```javascript
requiredAuth = (obj, params, context) => {
  return (controller) => {
    if (context.currentUser) // ...
    // ...
  }
}
```

Reply:
Sorry:
```javascript
requiredAuth = (controller) => {
  return (obj, params, context) => {
    if (context.currentUser) // ...
    // ...
  }
}
```

Reply:
Ummm ok ok